src/chef-mover/src/mover_global_installation_admins_callback.erl

```diff
@@ -0,0 +1,203 @@
+%% ex: ts=4 sw=4 et
+%% @author Tyler Cloke <tyler@chef.io>
+%% @copyright 2015 Chef Software, Inc.
+%%
+%% This migration sets up a global group called global-admins.
+%% It also grants READ and CREATE permissions on the users container
+%% to the global-admins group.
+%%
+-module(mover_global_installation_admins_callback).
+
+-export([
+         migration_init/0,
+         migration_complete/0,
+         migration_type/0,
+         supervisor/0,
+         migration_start_worker_args/2,
+         migration_action/2,
+         next_object/0,
+         error_halts_migration/0,
+         reconfigure_object/2,
+         needs_account_dets/0
+        ]).
+
+-include("mover.hrl").
+-include("mv_oc_chef_authz.hrl").
+
+-define(GLOBAL_PLACEHOLDER_ORG_ID, <<"00000000000000000000000000000000">>).
+
+-record(container, {authz_id}).
+-record(mover_chef_group, {
+          id,
+          org_id,	  
+          authz_id,
+          name,
+          last_updated_by,
+          created_at,
+          updated_at
+         }).
+
+
+migration_init() ->
+    mv_oc_chef_authz_http:create_pool(),
+    mover_transient_migration_queue:initialize_queue(?MODULE, [?GLOBAL_PLACEHOLDER_ORG_ID]).
+
+migration_action(GlobalOrgId, _AcctInfo) ->
+    SuperuserAuthzId = mv_oc_chef_authz:superuser_id(),                              
+    GlobalAdminAuthzId = case create_global_admins_authz_group(SuperuserAuthzId) of
+        AuthzId when is_binary(AuthzId) ->
+            AuthzId;
+        AuthzError ->
+            lager:error("Could not create new authz group for global-admins."),
+            throw(AuthzError)
+    end,
+    
+    case create_global_admins_global_group(GlobalAdminAuthzId, GlobalOrgId) of
+        {chef_sql, {GroupError, _}} ->
+	    lager:error("Could not create new erchef group for global-admins."),
+            throw(GroupError);
+	_ -> true
+    end,
+    
+    %% Grant global group permissions on users container 
+    UserContainerAuthzId = get_user_container_authz_id(),
+    Body = iolist_to_binary(jiffy:encode(
+                              {
+                                [
+                                 {<<"actors">>, []}, 
+                                 {<<"groups">>, [GlobalAdminAuthzId]}
+                                ]
+                              }
+                             )),
+
+    ReadUrl = make_url([containers, UserContainerAuthzId, read]),
+    case oc_chef_authz_http:request(ReadUrl, put, [], Body, SuperuserAuthzId) of
+        ok -> true;
+        {error, not_found} -> throw_not_found("read");
+        {error, server_error} -> throw_server_error()
+    end,
+
+    CreateUrl = make_url([containers, UserContainerAuthzId, create]),
+    case oc_chef_authz_http:request(CreateUrl, put, [], Body, SuperuserAuthzId) of
+        ok -> true;
+        {error, not_found} -> throw_not_found("create");
+        {error, server_error} -> throw_server_error()
+    end.
```

Comment:
Let's also add the pivotal user to this group as part of the migration. Maybe one day, we can move entirely to depending on this group instead of the specific pivotal user...